experiment_helpers/run_checkLPC.m

```diff
@@ -0,0 +1,107 @@
+function [expt, exptPre] = run_checkLPC(expt, exptPre)
+% Runs a short speech experiment using run_measureFormants_audapter, then
+% plots formant means using check_audapterLPC. Useful for determining LPC
+% order and vowel means for participants in most speech experiments. The
+% indended use is that this will happen near the beginning of a study,
+% before the main phase. 
+% 
+% The result of this function is saving a file `exptPre`. The variable 
+% exptPre can also be passed back as an output argument.
+%
+% This function assumes the input argument `expt` has the following fields:
+%   expt.dataPath
+%   expt.snum
+%   expt.gender
+
+% 2022-11 Chris Naber init.
+
+%% setup
+if nargin < 2
+    exptPre = struct;
+end
+
+defaultParams.snum = expt.snum;
+defaultParams.gender = expt.gender;
+defaultParams.dataPath = fullfile(expt.dataPath, 'pre');
+defaultParams.words = {'bid' 'bat' 'bed'};
+defaultParams.conds = {'noShift'};
+defaultParams.trackingFileLoc = 'experiment_helpers'; % default single-syllable, one word Audapter OST file
+defaultParams.trackingFileName = 'measureFormants';   % default single-syllable, one word Audapter OST file
+defaultParams.audapterParams.fb = 3;                  % feedback mode. 3=participant hears speech and noise
+if isfield(expt, 'bTestMode') && expt.bTestMode
+    defaultParams.nblocks = 2;  % number of repetitions of each word
+else
+    defaultParams.nblocks = 10; % number of repetitions of each word
+end
+defaultParams.amplcalc.checkMethod = 'mean';
+defaultParams.amplcalc.rmsThresh = 0; % don't automatically prompt pp to change volume during pretest phase
+% amplcalc.limits is a 2x2 array structured like this:
+%        [GoodLow, GoodHi;
+%         WarnLow, WarnHi]
+% In check_rmsThresh, a line is drawn in green between the low and hi Good limits,
+% and a line in yellow between the Warn limits.
+% With these limits, the green region is 80-86 dBA on SMNG hardware.
+% Warn region is +/- another 1.5 dBA.
+defaultParams.amplcalc.limits = [0.042, 0.060; 0.037, 0.065];
+
+% reconcile exptPre and defaultParams
+exptPre = set_missingFields(exptPre, defaultParams, 0);
+
+% set settings that require reconciled exptPre and defaultParams
+if ~isfield(exptPre, 'ntrials')
+    exptPre.ntrials = exptPre.nblocks * length(exptPre.words);
+end
```

Comment:
Just FYI, for this and the following, I think you can do:
```suggestion
exptPre = set_missingField(exptPre,'ntrials',exptPre.nblocks * length(exptPre.words));
```
(Not at all necessary but we do have the function as a shortcut!)

Reply:
Fixed this and the other instance to use `set_missingField`

Reply:
I can't see the fix here (I don't need it to approve the PR, but just FYI)